### DISCUSSION TITLE
default value in style expression

### DISCUSSION THREAD

**Discussion by RDUser-dev** [openlayers/openlayers#16131] [Q&A] [answered]
We are using a style expression to create a WebGLStyle an works well.
We use an expression like this `'stroke-width': ['get', 'strokeWidth']`
to read the strokeWidth property from the feature, but there may be cases where the property is not set yet.
How can we check if the property exists and possibly provide a fallback value?
It would have been great if the get operator was `['get', 'attributeName', typeHint, fallbackValue]` but according to the documentation https://openlayers.org/en/latest/apidoc/module-ol_style_expressions.html it is not.
Thanks to everyone for the help

**Comment by RDUser-dev** [comment]
I did some more research on github, and found these issues:
https://github.com/openlayers/openlayers/issues/15699
https://github.com/openlayers/openlayers/issues/15701
So if I understand correctly there is currently no way in WebGL to set the default for a property and there are no workarounds right?
Thanks

**Comment by jahow** [comment, marked as answer]
The solution would be to do: `['coalesce', ['get','propertyname'], 'default value']]`

Unfortunately this is not yet supported in WebGL renderers.

**Reply by RDUser-dev** [reply]
ok thanks for the confirmation, for now we are back to using canvas.
I think it is still early to switch to webgl
